wire-schema/src/test/java/com/squareup/wire/schema/PrunerTest.java

```diff
@@ -79,4 +79,105 @@
     assertThat(pruned.getType("RequestB")).isNull();
     assertThat(pruned.getType("ResponseB")).isNull();
   }
+
+  @Test public void retainField() throws Exception {
+    Schema schema = new SchemaBuilder()
+        .add("service.proto", ""
+            + "message MessageA {\n"
+            + "  optional string b = 1;\n"
+            + "  optional string c = 2;\n"
+            + "}\n"
+            + "message MessageB {\n"
+            + "}\n")
+        .build();
+    Schema pruned = schema.retainRoots(ImmutableList.of("MessageA#b"));
```

Comment:
Mostly because it doesn't scale to say what you want at this granularity but rather what you don't want. Or at least in our case it doesn't.